run.py

```diff
@@ -138,6 +148,8 @@ def _validate_devices(devices: str):
 if __name__ == "__main__":
     parser = argparse.ArgumentParser(__doc__)
     SUPPORT_DEVICE_LIST = ["cpu", "cuda"]
+    if (torch.backends.mps.is_available()):
```

Comment:
This looks good to me now, but later we may add a function to generate a list of available devices so it can be shared across all test scripts.

Reply:
Sounds good.